### PR TITLE
Add Neutron forking (rehearsal) documentation

### PR DESCRIPTION
**Pull request by albertandrejev**
This PR contains documentation about how to run Neutron Mainnet Fork for testing purposes.

### CODE REVIEW

File: docs/other_tools/rehearsal/overview.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# Overview
+
+# Neutron Mainnet Fork
+
+Neutron mainnet fork is the utilitiy to automate the creation and execution of a mainnet fork for the Neutron network, facilitating integration tests. With the rapid changes in the blockchain landscape, preliminary testing is crucial. Forking the mainnet allows developers to rigorously test contracts, modules, and other functionalities in an environment that mirrors the current mainnet. 
+
+Note ‼️: The fork is using neutron image with disabled signature verification, so you can send transactions on behalf of any account. This is done to simplify testing process. Please do not use this fork for any other purposes. Testing scripts has a [helper](https://github.com/hadronlabs-org/rehearsal/blob/main/src/libs/wallet.ts#L3) to allow you to send transactions on behalf of any account.
```
Comment (pr0n00gler) [inline docs/other_tools/rehearsal/overview.md:7]: Pls fix formatting of Note part as in the rest of the documentation

File: docs/other_tools/rehearsal/overview.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+# Hardware Requirements
+
+To ensure smooth operation with this project, the following hardware specifications are recommended:
+
+1. **Central Processing Unit (CPU):** Minimum of 4 cores; 8 cores are recommended.
+2. **Random Access Memory (RAM):** A minimum of 16GB, but 32GB or more is recommended for optimal performance.
+3. **Storage:** At least 20GB of free space on an SSD. An NVMe SSD is recommended for faster read/write operations.
+4. **Network:** A stable internet connection with a minimum download speed of 100 Mbps.
+
+Note: These requirements are based on the standard data volume processed by the Neutron network. More intensive operations or an increase in the blockchain's size may necessitate more robust hardware.
```
Comment (pr0n00gler) [inline docs/other_tools/rehearsal/overview.md:18]: Fix here note as well

File: docs/other_tools/rehearsal/overview.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+2. **Prototyping New Features:** New ideas and features can be rapidly prototyped and iterated upon using a fork, without disrupting the main network.
+3. **Testing Contracts:** Before deploying, smart contracts can be tested rigorously in an environment that simulates the real world, catching potential vulnerabilities.
+4. **Training and Education:** A fork provides an excellent environment for developers to learn and get accustomed to the network's nuances without any real-world consequences.
+5. **Debugging and Troubleshooting:** If any issues arise on the mainnet, they can be reproduced on the fork for a deeper analysis.
+
+This approach strengthens trust in the blockchain community and helps in preventing unforeseen problems when changes are made live.
+
+# Directory Structure
+
+- **`./snapshot` Directory:** This directory contains latest snapshot that was downloaded from https://snapshot.neutron.org/ (see `raw` snapshot). If you want to update snapshot simply delete contents of the `snapshot` directory and clear approprite docker volumes.
```
Comment (pr0n00gler) [inline docs/other_tools/rehearsal/overview.md:40]:
```suggestion
- **`./snapshot` Directory:** This directory contains latest snapshot that was downloaded from [Neutron Network Snapshots Service](https://snapshot.neutron.org)(see `raw` snapshot). If you want to update snapshot simply delete contents of the `snapshot` directory and clear appropriate docker volumes.
```

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

File: docs/other_tools/rehearsal/overview.md (hunk 1 above)
Comment (zavgorodnii) [reply docs/other_tools/rehearsal/overview.md:7]: Testing scripts **have** a

**Review by NeverHappened — APPROVED** [review]

**Review by oldremez — APPROVED** [review]

**Review by ratik — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**